### PR TITLE
fix(std/http): form valid URL if no hostname is passed to serve

### PR DESCRIPTION
**Pull request by 3bert**
This isn't causing any problems right now, but once proper host parsing has been implemented, calling `serve()` with a string argument without a hostname, e.g.
```
serve(':80')
```
will throw.

~~The proposed fix prepends a placeholder hostname to `addr` just to get the URL through the parser. The advantage this approach has over defining a default hostname in `_parseAddrFromStr()` is that `Deno.listen()`'s default hostname can be used. I think this is the right way to do this.~~ Doing this separately later.

To summarize: `serve()` intentionally accepts a host string which may be invalid as a URL's host, and that string (with `"http://"` prepended to it) is passed to `URL()`, which is currently too lax to reject it.

Thoughts?

### CODE REVIEW

**Comment by ry** [comment]
@3bert Would you wan to revert e92cf5b9e8530f7edf5cb7b157e6334a013da10d as well?

**Comment by 3bert** [comment]
@ry I see no reason to. This was introduced back in #5263 when address parsing started relying on `URL()`. To clarify, this is a preemptive patch; it currently causes no changes in behaviour.

**Comment by 3bert** [comment]
@ry Take #6698 first and I'll redo this one based on that. This is less important in the near-term. I can open a separate PR for not re-defining the default hostname when I have that ready.

**Comment by 3bert** [comment]
Resolved in another PR.
